Nrjwolf/unity-auto-attach-component-attributes
# Make repository upm compatible

### Summary
Make repository compatible with Unity Package Manager. Now can install it as a package (submodule installation option also remains).

### Details
Project structure is made similar to https://github.com/Nrjwolf/unity-ios-easy-native-alert following [official Unity package layout](https://docs.unity3d.com/Manual/cus-layout.html). Types are moved to `Nrjwolf.Attributes` namespace not to pollute global one. Yes, now to use attributes one must add `using Nrjwolf.Attributes;`, this is reflected in updated README.md

Package id is selected to be `com.nrjwolf.autoattachattributes`, open for discussion to change to `nrjwolf.games.autoattachattributes` though I prefer com. prefix.

### How to test
Install package via git using source branch:
```json
{
  "dependencies": {
    "com.nrjwolf.autoattachattributes": "https://github.com/rfadeev/unity-auto-attach-component-attributes.git#make-repository-upm-compatible"
  }
}
```
Observe package code compiles without warnings and can use auto attach attributes. Check project builds for some platform without errors.
